model/Index/Handler/ResourceUpdatedHandler.php

```diff
@@ -0,0 +1,195 @@
+    private $resolver;
+
+    public function __construct(
+        LoggerInterface $logger,
+        IndexDocumentBuilderInterface $indexDocumentBuilder,
+        SearchInterface $searchService,
+        ItemResourceSpecification $itemSpecification,
+        QtiService $qtiService,
+        ItemMediaResolver $resolver = null
+    ) {
+        parent::__construct(
+            $logger,
+            $indexDocumentBuilder,
+            $searchService,
+            [ResourceUpdated::class]
+        );
+
+        $this->itemSpecification = $itemSpecification;
+        $this->qtiService = $qtiService;
+        $this->resolver = $resolver;
+    }
+
+    protected function getResource(Event $event): core_kernel_classes_Resource
+    {
+        /** @var $event ResourceUpdated */
+        return $event->getResource();
+    }
+
+    /**
+     * @throws common_exception_InconsistentData
+     * @throws common_Exception
+     */
+    protected function doHandle(
+        Event $event,
+        core_kernel_classes_Resource $resource
+    ): void {
+        $doc = $this->getDocumentFor($resource);
+        $totalIndexed = $this->searchService->index([$doc]);
+
+        if ($totalIndexed < 1) {
+            $this->logResourceNotIndexed($resource, $totalIndexed);
+        }
+    }
+
+    /**
+     * @throws common_exception_InconsistentData
+     * @throws common_Exception
+     */
+    private function getDocumentFor(
+        core_kernel_classes_Resource $resource
+    ): IndexDocument {
+        $this->logger->debug(
+            sprintf(
+                '%s: Preparing data to index, resource = %s',
+                self::class,
+                $resource->getUri()
+            )
+        );
+
+        $document = $this->indexDocumentBuilder->createDocumentFromResource(
+            $resource
+        );
+
+        return new IndexDocument(
+            $document->getId(),
+            $this->getBody($document, $resource),
+            $document->getIndexProperties(),
+            $document->getDynamicProperties(),
+            $document->getAccessProperties()
+        );
+    }
+
+    /**
+     * @throws common_Exception
+     */
+    private function getBody(
+        IndexDocument $document,
+        core_kernel_classes_Resource $resource
+    ): array {
+        $references = [];
+
+        if ($this->itemSpecification->isSatisfiedBy($resource)) {
+            $references = $this->getResourceURIsFromItem($resource);
+        }
+
+        $body = $document->getBody();
+        $body['referenced_resources'] = $references;
+
+        return $body;
+    }
+
+    /**
+     * @throws common_Exception
+     */
+    private function getResourceURIsFromItem(
+        core_kernel_classes_Resource $resource
+    ): array {
+        $resolver = $this->resolver ?? $this->buildResolver($resource);
+        if ($resolver === null) {
+            return [];
+        }
+
+        $mediaURIs = [];
+        $qtiItem = $this->qtiService->getDataItemByRdfItem($resource);
```

Comment:
We can simplify by using `RdfMediaRelationRepository::findAllByTarget`

Reply:
Done :point_right: e919eb1620d94169ce903860274771efe3ab582c, thanks!